libbeat/filter/condition.go

```diff
@@ -11,247 +11,308 @@ import (
 	"github.com/elastic/beats/libbeat/logp"
 )
 
+type RangeValue struct {
+	gte *float64
+	gt  *float64
+	lte *float64
+	lt  *float64
+}
+
+type EqualsValue struct {
+	Int uint64
+	Str string
+}
+
 type Condition struct {
-	Equals   map[string]EqualsValue
-	Contains map[string]string
-	Regexp   map[string]*regexp.Regexp
-	Range    map[string]RangeValue
+	equals   map[string]EqualsValue
+	contains map[string]string
+	regexp   map[string]*regexp.Regexp
+	rangexp  map[string]RangeValue
```

Comment:
can I use the same key with different kind of expressions, but not with same kind of expression?

e.g.

```
condition:
  equals:
    http.code: 20
  range:
    http.code: ...
```

How about switching to:

```
type Condition struct {
    checks []FieldCheck
}

type FieldCheck struct {
    field string
    check Check
}

type Check interface {
    func CheckValue(v interface{}) (bool, error)
}

type IntEq struct { i int64 }
func (i IntEq) CheckValue(v interface{}) (bool, error) { ... }

type FloatEq struct { f float64 }
func (f FloatEq) CheckValue(v interface{}) (bool, error) { ... }

type StringEq struct { s string }
func (s StringEq) CheckValue(v interface{}) (bool, error) { ... }

func (c *Condition) Check(event common.MapStr) (bool, error) {
    for _, fieldCheck := range c.checks {
      val := extractValue(event, fieldCheck.field)
      if b, err := fieldCheck.check.Check(val); !b || err != nil {
          return b, err
      }
    }
    return true, nil
}
```

If one key is used multiple times, the terms should be or combined. For example:

```
condition:
  equals:
    http.code: 200
    http.code: 302
```

Will act like `http.code == 200 || http.code == 320`